src/renderer/components/notifications/notifications.store.tsx

```diff
@@ -18,6 +22,38 @@ export interface Notification {
   message: NotificationMessage;
   status?: NotificationStatus;
   timeout?: number; // auto-hiding timeout in milliseconds, 0 = no hide
+  onClose?(): void; // additonal logic on when the notification times out or is closed by the "x"
+}
+
+export interface MainNotification {
+  title: string;
+  body: string;
+  buttons?: ({
+    backchannel: string;
+  } & ButtonProps)[];
+  status: NotificationStatus;
+  timeout?: number;
+  closeChannel?: string;
+}
+
+function RenderButtons({ id, buttons }: { id: string, buttons?: MainNotification["buttons"] }) {
+  if (!buttons) {
+    return null;
+  }
+
+  return (
+    <>
+      <br />
+      <div className="ButtonPannel flex row align-right box grow">
+        {buttons.map(({ backchannel, ...props }) => (
+          <Button key={backchannel} {...props} onClick={() => {
+            ipcRenderer.send(backchannel);
```

Comment:
Does this mean that it's only possible to send notifications from `main`?  (`broadcastMessage()` probably would solve this)

Reply:
I don't understand, `renderer` already has a way to display notifications.

Reply:
But that's different way, right? Would it make sense to have just one way to trigger notifications (might be out of scope of this PR).

Reply:
I think that is out of scope for this PR. Yes it is a different way. They call `Notification.(info|error|success)(...)`

Reply:
Related to my other review comment (https://github.com/lensapp/lens/pull/1941#r565863019)): could we just use `Notification` component to show a notification in `renderer`? If no -> could we enhance it so that it can handle these kind of use cases. We really should have just one way to trigger notifications (note: `Notification` is also exposed via extension API).